iam-login-service/src/main/webapp/resources/iam/apps/dashboard-app/components/clients/client/client.component.js

```diff
@@ -16,6 +16,49 @@
 (function () {
     'use strict';
 
+    function ClientSecretViewController($uibModal, $uibModalInstance, toaster, ClientsService, data) {
+        var $ctrl = this;
+        $ctrl.data = data;
+        $ctrl.isNewClient = data.isNewClient;
+        $ctrl.newClient = data.client;
+        $ctrl.secret = $ctrl.newClient.client_secret;
+        $ctrl.clientId = $ctrl.newClient.client_id;
+        $ctrl.showSecret = false;
+        $ctrl.confirmation = true;
+
+        self.clipboardSuccess = clipboardSuccess;
+        self.clipboardError = clipboardError;
+      
+        $ctrl.ok = function() {
+            $uibModalInstance.close($ctrl.selected);
+        };
+      
+        $ctrl.closeModal = function() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        $ctrl.toggleSecretVisibility = function() {
+            $ctrl.showSecret = !$ctrl.showSecret;
+        };
+
+        function clipboardError(event) {
+            toaster.pop({
+                type: 'error',
+                body: 'Could not copy secret to clipboard!'
+            });
+        }
+
+        function clipboardSuccess(event, source) {
+            toaster.pop({
+                type: 'success',
+                body: 'Secret copied to clipboard!'
+            });
+            event.clearSelection();
+            if (source === 'secret') {
```

Comment:
At line 54 you already assumed that the event source is of type `secret`. Why this conditions is after the notification popup dispatch?.
I think that everything or anything should be inside inside the condition